### PR TITLE
Fix kafka connector issues with latest ballerina changes

### PR DESCRIPTION
**Pull request by daneshk**
## Purpose
This PR is to fix kafka connector issues with latest ballerina changes.

### CODE REVIEW

File: component/src/main/wso2/kafka/consumer.bal [hunk 1]
```diff
@@ -188,125 +189,140 @@ public type SimpleConsumer object {
...
+    # Returns current subscription of topics for a consumer.
+    #
+    # + return - subscribed topic array for consumer if executes successfully, error otherwise.
     public extern function getSubscription() returns string[]|error;
 
-    documentation { Returns current subscription of topics for a consumer
-        P{{partition}} partition in which offset is returned for consumer
-        R{{}} committed offset for consumer for given partition
-    }
+    # Returns current subscription of topics for a consumer.
```
Comment (ThisaruGuruge) [inline component/src/main/wso2/kafka/consumer.bal]: This returns committed offset of a consumer, for a given partition. The definition is duplicated from the above, not changed.

File: component/src/main/wso2/kafka/producer.bal [hunk 2]
```diff
@@ -79,82 +79,83 @@ public type ProducerConfig record {
...
-        P{{value}} record contents
-        P{{topic}} topic the record will be appended to
-        P{{key}} key that will be included in the record
-        P{{partition}} partition to which the record should be sent
-        P{{timestamp}} timestamp of the record, in milliseconds since epoch
-    }
+    # Simple Send action which produce records to Kafka server.
+    #
+    # + value - record contents.
+    # + topic - topic the record will be appended to.
```
Comment (ThisaruGuruge) [inline component/src/main/wso2/kafka/producer.bal]: Better if we state this as `topic - topic to which the record will be appended.

File: component/src/main/wso2/kafka/producer.bal [hunk 3]
```diff
@@ -79,82 +79,83 @@ public type ProducerConfig record {
...
+    # Close action which closes Kafka producer.
     public extern function close();
 
-    documentation { GetTopicPartitions action which returns given topic partition information
-        P{{topic}} topic which partition information is given
-        R{{}} partition for given topic
-    }
+    # GetTopicPartitions action which returns given topic partition information.
+    # + topic - topic which partition information is given.
+    # + returns - partition for given topic.
```
Comment (ThisaruGuruge) [inline component/src/main/wso2/kafka/producer.bal]: This should be `partition(s) for the given topic`

File: component/src/main/wso2/kafka/consumer.bal [hunk 4]
```diff
@@ -188,125 +189,140 @@ public type SimpleConsumer object {
...
+    # Subscribes to consumer to external Kafka broker topic pattern.
+    #
+    # + regex - topic pattern to be subscribed.
+    # + returns - Returns an error if encounters an error, returns nil otherwise.
     public extern function subscribeToPattern(string regex) returns error?;
 
-    documentation { Subscribes to consumer to external Kafka broker topic array
-        P{{topics}}tTopic array to be subscribed
-    }
+    # Subscribes to consumer to external Kafka broker topic array.
```
[the remark this answers is not in the document]
Comment (daneshk) [reply component/src/main/wso2/kafka/consumer.bal]: Fixed in 037ee71

File: component/src/main/wso2/kafka/consumer.bal [hunk 5]
```diff
@@ -188,125 +189,140 @@ public type SimpleConsumer object {
...
+    # + topics - tTopic array to be subscribed.
+    # + returns - Returns an error if encounters an error, returns nil otherwise.
     public extern function subscribe(string[] topics) returns error?;
 
-    documentation { Subscribes to consumer to external Kafka broker topic with rebalance listening is enabled
-        P{{topics}} topic array to be subscribed
-        P{{onPartitionsRevoked}} function will be executed if partitions are revoked from this consumer
-        P{{onPartitionsAssigned}} function will be executed if partitions are assigned this consumer
-    }
+    # Subscribes to consumer to external Kafka broker topic with rebalance listening is enabled.
```
[the remark this answers is not in the document]
Comment (daneshk) [reply component/src/main/wso2/kafka/consumer.bal]: Fixed in 037ee71

File: component/src/main/wso2/kafka/consumer.bal (hunk 1 above)
Comment (daneshk) [reply component/src/main/wso2/kafka/consumer.bal]: Fixed in 037ee71

File: component/src/main/wso2/kafka/producer.bal (hunk 3 above)
Comment (daneshk) [reply component/src/main/wso2/kafka/producer.bal]: Fixed in 8e52317

File: component/src/main/wso2/kafka/producer.bal (hunk 2 above)
Comment (daneshk) [reply component/src/main/wso2/kafka/producer.bal]: Fixed in 8e52317

**Comment by daneshk** [comment]
Closing this PR, since this fix contains in PR #18

**Pull request closed without merging**